addons/web_editor/static/lib/odoo-editor/src/commands/shiftEnter.js

```diff
@@ -60,9 +61,14 @@ HTMLAnchorElement.prototype.oShiftEnter = function () {
     }
     if (brs.includes(firstChild)) {
         brs.forEach(br => anchor.before(br));
-        setSelection(...rightPos(brs[brs.length - 1]));
     } else if (brs.includes(lastChild)) {
+        const brToRemove = isBlock(anchor) ? brs.shift() : null;
         brs.forEach(br => anchor.after(br));
-        setSelection(...rightPos(brs[0]));
+        if (brToRemove) {
+            brToRemove.remove();
+            setSelection(...leftPos(brs[0]));
+        } else {
+            setSelection(...rightPos(brs[0]));
+        }
```

Comment:
```suggestion
        const brToRemove = isBlock(anchor) && brs.pop();
        brs.forEach(br => anchor.after(br));
        if (brToRemove) {
            // When the anchor tag is block, keeping the two `br` tags
            // would have resulted into two new lines instead of one.
            brToRemove.remove();
            setSelection(...leftPos(brs[0]));
        } else {
            setSelection(...rightPos(brs[0]));
        }
```
Note: Among other things which don't change the effect at all, I changed `shift` to `pop` because I'm always confused between `shift` and `unshift` and I figured removing the first or last BR would not change anything in practice. Feel free to use `shift` in the suggestion instead if in fact I'm wrong and it has an impact.